src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -368,10 +370,10 @@ private BoundExpression SynthesizeCall(CSharpSyntaxNode syntax, BoundExpression
 
         /// <summary>
         /// Lower a foreach loop that will enumerate the characters of a string.
-        /// 
+        ///
         /// string s = x;
         /// for (int p = 0; p &lt; s.Length; p = p + 1) {
-        ///     V v = (V)s.Chars[p];
+        ///     V v = (V)s.Chars[p];   /* OR */   (D1 d1, ...) = (V)s.Chars[p];
```

Comment:
Do you mean an empty string specifically, or a string in general?

I do have a test for iterating over the characters in a string:
```C#
      public void ForeachStringDeclarationWithNesting()
        {
            string source = @"
class C
{
    static void Main()
    {
        foreach ((int x1, var (x2, x3)) in M())
        {
            System.Console.Write(x1 + "" "" + x2 + "" "" + x3 + "" - "");
        }
    }
    static string M() { return ""12""; }
```